libvips/iofuncs/threadpool.c

```diff
@@ -355,24 +362,33 @@ get_num_processors( void )
 {
 	/* Count the CPUs currently available to this process.  
 	 */
+	SYSTEM_INFO sysinfo;
 	DWORD_PTR process_cpus;
 	DWORD_PTR system_cpus;
 
+	/* This *never* fails, use it as fallback 
+	 */
+	GetNativeSystemInfo( &sysinfo );
+	nproc = (int) sysinfo.dwNumberOfProcessors;
+
 	if( GetProcessAffinityMask( GetCurrentProcess(), 
 		&process_cpus, &system_cpus ) ) {
-		unsigned int count;
+		unsigned int af_count;
 
-		for( count = 0; process_cpus != 0; process_cpus >>= 1 )
+		for( af_count = 0; process_cpus != 0; process_cpus >>= 1 )
 			if( process_cpus & 1 )
-				count++;
+				af_count++;
 
-		if( count > 0 )
+		/* Prefer affinity-based result, if available 
+		 */
+		if( af_count > 0 )
 			nproc = count;
```

Comment:
Does this line need to be changed too?
```diff
- nproc = count;
+ nproc = af_count;
```

Reply:
Yes, it had to be changed too. Fixed with https://github.com/libvips/libvips/pull/1590/commits/ceacd0bab9296f02d79c3ec07f8e6fcb8ff383f9.

Thanks!